pkg/flow/module.go

```diff
@@ -0,0 +1,127 @@
+package flow
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"sync"
+
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/cluster"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/web/api"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type module struct {
+	mut sync.Mutex
+	f   *Flow
+	o   *moduleOptions
+}
+
+type moduleOptions struct {
+	ID     string
+	export component.ExportFunc
+	*moduleControllerOptions
+}
+
+var (
+	_ component.Module = (*module)(nil)
+)
+
+// newModule creates a module instance for a specific component.
+func newModule(o *moduleOptions) *module {
+	return &module{
+		o: o,
+	}
+}
+
+// LoadConfig parses River config and loads it.
+func (c *module) LoadConfig(config []byte, args map[string]any) error {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if c.f == nil {
+		f := New(Options{
```

Comment:
It's not obvious to me, but where is the unwrapping of logging/tracing/metrics happening? Is that not done yet?